ush/config_defaults.yaml

```diff
@@ -842,6 +842,10 @@ workflow:
   #   the suffix "_oldNNN", where NNN is a 3-digit integer chosen to make
   #   the new name unique.
   #
+  # * "upgrade":
```

Comment:
@danielabdi-noaa This description is somewhat unclear to me.  So if "upgrade" is specified, I see below that a copy is made using `rsync`, but then what is upgraded?  What does it mean to "upgrade a directory"?  Also, you mention `EXPTDIR`, but it could be any directory, right?  Can you clarify a bit?  Thanks.

Reply:
@gsketefian Here is a better description of the "upgrade" feature. I should probably add the full comment from there in this PR.  
https://github.com/NOAA-GSL/regional_workflow/blob/feature/RRFS_dev1/ush/bash_utils/check_for_preexist_dir_file.sh#L112-L152

The implementation in `RRFS_dev1` checks whether the directory whose existence we are checking is EXPTDIR or not, and then behaves differently accordingly. If it is an EXPTDIR it is equivalent to the "rename" option. In other cases it keeps the same directoy and just does `rsync` instead. This feature is apparently useful to keep rocoto jobs running but one can choose to use "rename" in all cases.

Note in SRW we have to version of `check_preexisting_dir` (python and bash version). The python version is explcitily used for renaming EXPTDIR so we don't need to check it it is another directory, so it effectively does "rename". The bash version is used in ex-scripts/j-jobs where we don't rename EXPTDIR so it behaves like an upgrade/"don't do nothing to the directory".

Reply:
@danielabdi-noaa Thanks for the explanation.  It's much clearer now.  Some additional questions:
1) It seems like in the python script, what the "upgrade" option does is to make a backup copy of the directory while keeping the original.  Is that correct?  If so:
  a) Wouldn't it suffice to just use `cp` instead of `rsync`?
  b) Why not call the option something like "backup" instead of "upgrade" since "upgrade" sounds like something is being added to/updated in the original directory, which it isn't afa I can tell?
2) Why keep both python and shell scripts?  Why not just keep the python (and create a shell wrapper if necessary), and call it with "upgrade" (or "backup") when operating on `EXPTDIR` and with "delete" when operating on "grid", "orog", etc (or not call the script at all for "grid", "orog", ...)?

Reply:
@danielabdi-noaa 
1) I suppose it can be reduced to `cp` in most cases, but maybe they wanted to add syncing experiment directory from remote machine? Will have to ask the original author about that. About the naming, yes the same directory contents are kept as before in both EXPTDIR and other folder types, so the name "upgrade"/"update" was probably meant to reflect what happens after that. I personally like "reuse" as the name, let me know what you think about it?

2) I don't like keeping around two versions of the same scripts (bash and python) either. Maybe we can move to using only the python script in a future PR?

Reply:
@danielabdi-noaa Thanks again for the explanations.  So:
1) "reuse" sounds good to me.
2) Do you mind creating an issue to remove the shell version of the script?  I just want to make sure we don't forget about it.  

Otherwise looks good, thanks.

Reply:
@gsketefian Thanks, I have updated the name from "upgrade" to "reuse", and also created an issue #696 to address the duplicate shell/python functionality.